src/status_im/contexts/wallet/wallet_connect/utils/typed_data_test.cljs

```diff
@@ -0,0 +1,53 @@
+(ns status-im.contexts.wallet.wallet-connect.utils.typed-data-test
+  (:require
+    [cljs.test :refer-macros [deftest is testing]]
+    [status-im.contexts.wallet.wallet-connect.utils.typed-data :as sut]))
+
+(deftest flatten-data-test
+  (testing "correctly returns the fallback when key is not string/keyword/number"
+    (is (= (sut/flatten-data {[1 2] "value"})
+           [[["unsupported-key"] "value"]])))
+
+  (testing "correctly flattens a simple map"
+    (is (= (sut/flatten-data {:zero 0
+                              "one" 1
+                              2     2})
+           [[["zero"] 0]
+            [["one"] 1]
+            [["2"] 2]])))
+
+  (testing "correctly flattens a simple vec"
+    (is (= (sut/flatten-data ["zero" "one"])
```

Comment:
The expected argument should be the first.

Example output from CIDER from another project:

```clojure
(deftest some-test
  (is (= 1 0)))
```

<img src="https://github.com/user-attachments/assets/a3df9f95-98f7-4422-b7a2-f54abd43edc0" width="350" />